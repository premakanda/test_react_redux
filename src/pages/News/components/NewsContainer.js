import React from 'react'
import { connect } from 'react-redux'
import NewsList from './NewsList'
import { getAll } from '../selectors'
import { getNews } from '../actions'

export class NewsContainer extends React.Component {
  componentDidMount() {
    this.props.onGetNews()
  }

  render() {
    const { news: { isLoading, data, errorMsg } } = this.props
    return (
      <React.Fragment>
        {errorMsg && <p>{errorMsg}</p>}
        {isLoading ? <p>Loading...</p> : null}
        {data && data.length && <NewsList data={data} />}
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => ({
  news: getAll(state),
})

const mapDispatchToProps = dispatch => ({
  onGetNews: () => dispatch(getNews()),
})

export default connect(mapStateToProps, mapDispatchToProps)(NewsContainer)
