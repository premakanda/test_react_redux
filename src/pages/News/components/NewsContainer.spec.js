import React from 'react'
import { shallow } from 'enzyme'
import { NewsContainer } from './NewsContainer'

describe('News container', () => {
  const props = {
    news: {
      data: [],
      isLoading: false,
      errorMsg: null,
    },
    onGetNews: () => {},
  }

  describe('News container initial', () => {
    const mockFetchGetNews = jest.fn()
    const nextProps = {
      ...props,
      onGetNews: mockFetchGetNews,
    }
    const newsContainer = shallow(<NewsContainer {...nextProps} />)

    it('renders properly', () => {
      expect(newsContainer).toMatchSnapshot()
    })

    it('dispatches the `onGetNews()` method it receives from props', () => {
      expect(mockFetchGetNews).toHaveBeenCalledTimes(1)
    })
  })

  describe('News container isLoading', () => {
    const nextProps = {
      ...props,
      news: {
        ...props.news,
        isLoading: true,
      },
    }

    const newsContainer = shallow(<NewsContainer {...nextProps} />)

    it('renders properly', () => {
      expect(newsContainer).toMatchSnapshot()
    })

    it('renders preloader', () => {
      expect(newsContainer.find('p').text()).toEqual('Loading...')
    })
  })

  describe('News container render <NewsList />', () => {
    const nextProps = {
      ...props,
      news: {
        ...props.news,
        data: [1],
      },
    }

    const newsContainer = shallow(<NewsContainer {...nextProps} />)

    it('renders properly', () => {
      expect(newsContainer).toMatchSnapshot()
    })

    it('renders <NewsList /> template', () => {
      expect(newsContainer.find('NewsList')).toHaveLength(1)
    })
  })

  describe('News container with errorMsg', () => {
    const nextProps = {
      ...props,
      news: {
        ...props.news,
        errorMsg: 'Something going wrong',
      },
    }

    const newsContainer = shallow(<NewsContainer {...nextProps} />)

    it('renders properly', () => {
      expect(newsContainer).toMatchSnapshot()
    })

    it('renders errorMsg', () => {
      expect(newsContainer.find('p').text()).toEqual(nextProps.news.errorMsg)
    })
  })
})
