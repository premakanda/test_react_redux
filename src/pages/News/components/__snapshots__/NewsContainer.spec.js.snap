// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News container News container initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    news={
      Object {
        "data": Array [],
        "errorMsg": null,
        "isLoading": false,
      }
    }
    onGetNews={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        null,
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      0,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          null,
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        0,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`News container News container isLoading renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    news={
      Object {
        "data": Array [],
        "errorMsg": null,
        "isLoading": true,
      }
    }
    onGetNews={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <p>
          Loading...
        </p>,
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      0,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <p>
            Loading...
          </p>,
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        0,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`News container News container render <NewsList /> renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    news={
      Object {
        "data": Array [
          1,
        ],
        "errorMsg": null,
        "isLoading": false,
      }
    }
    onGetNews={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        null,
        <NewsList
          data={
            Array [
              1,
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [
            1,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          null,
          <NewsList
            data={
              Array [
                1,
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [
              1,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`News container News container with errorMsg renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    news={
      Object {
        "data": Array [],
        "errorMsg": "Something going wrong",
        "isLoading": false,
      }
    }
    onGetNews={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <p>
          Something going wrong
        </p>,
        null,
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something going wrong",
        },
        "ref": null,
        "rendered": "Something going wrong",
        "type": "p",
      },
      null,
      0,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p>
            Something going wrong
          </p>,
          null,
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something going wrong",
          },
          "ref": null,
          "rendered": "Something going wrong",
          "type": "p",
        },
        null,
        0,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
